### PR TITLE
implement capability to write metadata and nos data to S3 bucket

### PR DESCRIPTION
**Pull request by ted80810**

### CODE REVIEW

**Review by galengorski — APPROVED** [review]
This looks good to me, I'll defer to @amsnyder for the s3 syntax though

File: 01_fetch/src/fetch_noaa_nos.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import urllib
 import requests, json
 
-def fetch_metadata(station_id, metadata_outfile):
+def fetch_metadata(station_id, metadata_outfile, bucket, write_file):
```
Comment (amsnyder) [inline 01_fetch/src/fetch_noaa_nos.py:7]:
```suggestion
def fetch_metadata(station_id, metadata_outfile, bucket, write_location):
```

File: 01_fetch/src/fetch_noaa_nos.py [hunk 2]
```diff
@@ -15,13 +15,26 @@ def fetch_metadata(station_id, metadata_outfile):
     metadata = df.iloc[1:,1:].rename(columns={df.columns[1]:"data_property",
                                               df.columns[2]:(station_id)}).replace(".self","", regex=True)
     metadata.to_csv(metadata_outfile, index=False)
+    if write_location == 'S3':
+        print('uploading to s3')
+        s3_client.upload_file(metadata_outfile, bucket, '01_fetch/out/'+os.path.basename(metadata_outfile))
 
-def fetch_noaa_nos_data(start_dt, end_dt, datum, station_id, time_zone, product, units, file_format, data_outfile):
+def fetch_noaa_nos_data(start_dt, end_dt, datum, station_id, time_zone, product, units, file_format, data_outfile, bucket, write_file):
```
Comment (amsnyder) [inline 01_fetch/src/fetch_noaa_nos.py:22]:
```suggestion
def fetch_noaa_nos_data(start_dt, end_dt, datum, station_id, time_zone, product, units, file_format, data_outfile, bucket, write_location):
```

File: 01_fetch/src/fetch_noaa_nos.py [hunk 3]
```diff
@@ -15,13 +15,26 @@ def fetch_metadata(station_id, metadata_outfile):
...
+        print('uploading to s3')
+        s3_client.upload_file(data_outfile, bucket, '01_fetch/out/'+os.path.basename(data_outfile))
 
 def main():
+    write_location: 'S3'
+    '''choose where you want to write your data outputs: local or S3'''
+    aws_profile: 'dev'
+    '''set name of AWS profile storing credentials for S3'''
+    s3_bucket: 'drb-estuary-salinity'
+    '''set AWS bucket to read/write to'''
```
Comment (amsnyder) [inline 01_fetch/src/fetch_noaa_nos.py:36]: - we want write location to be local by default...we can change it to S3 when we are actually ready to upload
-  moving comments above the line of code they are describing
- set up s3 client for upload

```suggestion
    '''choose where you want to write your data outputs: local or S3'''
    write_location: 'local'
    '''set name of AWS profile storing credentials for S3'''
    aws_profile: 'dev'
    '''set AWS bucket to read/write to'''
    s3_bucket: 'drb-estuary-salinity'
    '''set up AWS client'''
    s3_client = utils.prep_write_location(write_location, aws_profile)
```

**Review by amsnyder — CHANGES_REQUESTED** [review]
Looks good, but I made a few updates. There are two other changes you need to make to make this code run:
- your fetch data function name was changed, so the name needs to be updated when you execute it in main()
- add the new function inputs (s3_bucket and write_location) as inputs to your fetch function in main() where they are executed

It's always a good idea to actually try running the code to make sure it runs before submitting the pull request because I know I am not necessarily testing out the code to see that it works on my local computer!

**Pull request merged**